### ISSUE TITLE
[NUXE-552] Add Text Highlighting on React Typeahead kit 

### ISSUE THREAD

**Issue by andrew-k9** [powerhome/playbook#1389]
## Feature

As a developer, I'd like a playbook solution in the `Typeahead` field to text highlighting so that I can add fuzzy highlighting to a static array of users

![typeahead_highlight](https://user-images.githubusercontent.com/20051541/110954847-f6ea4700-8316-11eb-82c8-ee9df5c911b1.png)

## Reasoning

For my specific use-case, in Flight Control we have a typeahead field to select the users. On that field, [we're getting all the results before the component renders](https://github.com/powerhome/nitro-web/blob/8fd78e0caf6b7c19dd205820df3a5857f4763380/components/flight_control/app/components/FlightControl/columns/EditForm.jsx#L56) so that we can get a specific subset of users based off fields that aren't query-able on the user object. If they were, we would be able to use [the `nitro_react` `UserTypeahaed` kit, but that specifies an endpoint that will be queryed](https://github.com/powerhome/nitro-web/blob/8fd78e0caf6b7c19dd205820df3a5857f4763380/components/nitro_react/app/components/UserTypeahead.jsx#L135) (also I'm not 100% sure if that's even the right highlighting)

TL;DR: We need text highlighting in a typeahead field that does not query while typing and can't find good existing examples/ implementations

## Desired Solution

A boolean prop on the `Typeahead` component that's a flag for having the returned list of fields have fuzzy highlighting, which would be set to `false` when no prop given. This would make it not be a breaking change

## Alternatives we've tried

At this point, we would either 1) implement our own brittle solution to the problem or 2) copy and paste from the Connect/ Apm implementation of highlighting and patch it to our current needs, which at that point it is probably better to implement this in playbook directly 😅

If there are any alternatives by using some funky js that I haven't considered or implementations elsewhere that do this please let me know!

## Additional Info

- [CHPN story that sparked this discussion](https://nitro.powerhrg.com/runway/backlog_items/CHPN-189)
- [BOP are running into a similar issue going to react now](https://nitro.powerhrg.com/runway/backlog_items/BOP-380)
- This would _probably_ be something to add in the Rails side as well?

**Comment by kevinschlabach** [comment]
Birds of Prey is running into a similar issue on our work.... we are converting a Rails page to React and leveraging the same kit Andrew is mention here...

<img width="1149" alt="runway-team-list" src="https://user-images.githubusercontent.com/44237304/110965600-57cb4c80-8322-11eb-93f7-9585005b0ff7.png">

We also don't have an easy way to do this, and we probably shouldn't have every team writing their own javascript to solve this problem at each local page...   

This is a second vote for this kit improvement.  (And we discussed it with @evalouderback during our team meetings)

**Comment by jasperfurniss** [comment]
I added this to our backlog. https://nitro.powerhrg.com/runway/backlog_items/NUXE-552

**Comment by thestephenmarshall** [comment]
This will be added to an upcoming sprint

**Comment by thestephenmarshall** [comment]
Creating custom options is possible without changes to the Typeahead kit itself. That is because the library we are using, `react-select`, [allows for individual component replacement](https://react-select.com/components#replaceable-components). Here is an example of how this can be achieved now: https://codesandbox.io/s/typeahead-playbook-highlight-zj92j

**Comment by co1in-1ewis** [comment]
This feature request was merged and shipped with Playbook 11.19.0. Closing this issue.
